expat/fuzz/xml_parse_fuzzer.c

```diff
@@ -47,18 +47,45 @@ end(void *userData, const XML_Char *name) {
   (void)name;
 }
 
-int
-LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
-  XML_Parser p = XML_ParserCreate(xstr(ENCODING_FOR_FUZZING));
-  assert(p);
-
+static void
+ParseOneInput(XML_Parser p, const uint8_t *data, size_t size) {
   // Set the hash salt using siphash to generate a deterministic hash.
   struct sipkey *key = sip_keyof(hash_key);
   XML_SetHashSalt(p, (unsigned long)siphash24(data, size, key));
 
   XML_SetElementHandler(p, start, end);
   XML_Parse(p, (const XML_Char *)data, size, 0);
-  XML_Parse(p, (const XML_Char *)data, size, 1);
+  if (XML_Parse(p, (const XML_Char *)data, size, 1) == XML_STATUS_ERROR) {
+      XML_ErrorString(XML_GetErrorCode(p));
+      XML_GetCurrentLineNumber(parser);
```

Comment:
`parser` would need to be `p` here [to work](https://github.com/libexpat/libexpat/actions/runs/7458547770/job/20306703388?pr=797#step:6:386).

Reply:
PS: Also, we should be able to call `XML_GetCurrentLineNumber(p);` in the non-error case also. Let's not limit it to the error case then for more coverage?

Reply:
Indeed, fixing this

Reply:
At least, it showed CI tests the fuzzers build ;-)